test/SlowTests/Server/RecordingTransactionOperationsMergerTests.cs

```diff
@@ -0,0 +1,509 @@
+            //            using (var reader = new StreamReader(zippedStream))
+            //            {
+            //                var fileContent = File.ReadAllText(filePath);
+            //                JToken.Parse(fileContent);
+            //            }
+            var fileContent = File.ReadAllText(filePath);
+            JToken.Parse(fileContent);
+        }
+
+        [Fact]
+        public void RecordingDeleteCommand()
+        {
+            var filePath = Path.GetTempFileName();
+
+            var expectedNames = new[] { "Avi" };//, "Baruch", "Charlotte", "Dylan", "Eli", "Fabia", "George", "Harper" };
+            var users = expectedNames.Select(n => new User { Name = n }).ToArray();
+            User[] documents = { };
+
+
+            //Recording
+            using (var store = GetDocumentStore())
+            {
+                store.Maintenance.Send(new StartTransactionsRecordingOperation(filePath));
+                using (var session = store.OpenSession())
+                {
+                    foreach (var user in users)
+                    {
+                        session.Store(user);
+                    }
+                    session.SaveChanges();
+
+                    store.Commands().Execute(new DeleteDocumentCommand(users.First().Id, null));
+                }
+                store.Maintenance.Send(new StopTransactionsRecordingOperation());
+            }
+
+            //Replay
+            using (var store = GetDocumentStore())
+            using (var replayStream = new FileStream(filePath, FileMode.Open))
+            {
+                store.Maintenance.Send(new ReplayTransactionsRecordingOperation(replayStream));
+                using (var session = store.OpenSession())
+                {
+                    documents = session.Query<User>().ToArray();
+                }
+            }
+
+            Assert.Empty(documents);
+        }
+
+        [Theory]
+        [InlineData("Avi")]
+        [InlineData("Avi", "Baruch", "Charlotte", "Dylan", "Eli", "Fabia", "George", "Harper")]
+        public void RecordingPatchWithParametersByQuery(params string[] names)
+        {
+            var filePath = Path.GetTempFileName();
+
+
+            var users = names.Select(n => new User { Name = n }).ToArray();
+
+            const string AgeKey = "age";
+            const int newAge = 34;
+
+            //Recording
+            using (var store = GetDocumentStore())
+            {
+                store.Maintenance.Send(new StartTransactionsRecordingOperation(filePath));
+                using (var session = store.OpenSession())
+                {
+                    foreach (var user in users)
+                    {
+                        session.Store(user);
+                    }
+
+                    session.SaveChanges();
+                }
+
+                var query = $"FROM Users UPDATE {{  this.Age = ${AgeKey} ; }}";
```

Comment:
Let's avoid concat of queries, even in test. Use parameters.

Reply:
done